### DISCUSSION TITLE
Integration of Orthanc with MonaiLabel (Docker)

### DISCUSSION THREAD

**Discussion by emi-dm** [Project-MONAI/MONAILabel#1628] [Q&A]
Hi guys! Do you know if any tutorial or documentation exists to integrate Monai Label (Docker) with Orthanc (Docker)?

Thanks in advance!

**Comment by diazandr3s** [comment]
Hi @emi-dm,

Sorry, I'm catching up now on this discussion. Have you tried adding these lines: https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/ohif#installing-orthanc-dicomweb to the [MONAI Label Docker container](https://github.com/Project-MONAI/MONAILabel/blob/main/Dockerfile)?

In a previous MONAI Label version (i.e. 0.5.0) we had Orthanc and OHIF integrated: https://github.com/Project-MONAI/MONAILabel/blob/0.5.0/Dockerfile#L38

Also, I've been working with HuggingFace (HF) devs to have an HF Space where you have everything: MONAI Label server, OHIFv3 and Orthanc (DICOMWeb). Stay tuned! We'll publish this soon.

Hope this helps,

**Reply by emi-dm** [reply]
Hi @diazandr3s , thanks so much for taking the time to join the discussion! I was able to connect them by getting Orthanc up in a docker and Monai Label in a conda environment. Do you propose to install Orthanc interactively inside the Monai Label container? 

Thank you very much for the quick reply! Regarding HF, it is very crazy, I think it will give a lot of opportunities to people who publish models related to medical imaging. I'm anxious to see the news, please let me know!

**Reply by diazandr3s** [reply]
Hi @emi-dm,

> I was able to connect them by getting Orthanc up in a docker and Monai Label in a conda environment.

This is great! That's another way of connecting both. No need to have Orthanc on the same Docker container.

> Regarding HF, it is very crazy, I think it will give a lot of opportunities to people who publish models related to medical imaging. I'm anxious to see the news, please let me know!

Yes, I think the community will benefit a lot from this integration. Hopefully this or early next week we should have the first release publicly available.

Thanks!

**Reply by sscotti** [reply]
I have been out of the loop on this for awhile.  Revisiting now that Digital Ocean has "PaperSpace" and GPU's.  Is there an existing docker-compose.yml / Docker Package that already has OHIF configured for MONAI label ?  I am also interested now in testing out the MONAI Model Zoo as well.

I would like to setup a really basic configuration on PaperSpace so that I can get some experience using models and then also training a new model.

I would like to use Orthanc mostly to just store datasets, and also sort of as a router and orchestrator for some of the work flows. 

I am anxious to test out the Model Zoo because I'd like to see what it returns as results.

**Reply by diazandr3s** [reply]
Hi @sscotti,

Thanks for commenting here. 

Regarding this:

> Is there an existing docker-compose.yml / Docker Package that already has OHIF configured for MONAI label ?

The MONAI Label container already comes with OHIFv3 out of the box: https://github.com/Project-MONAI/MONAILabel/blob/main/Dockerfile#L27

Just start the container like this: 

`docker run --gpus all --rm -ti --ipc=host --net=host projectmonai/monailabel:latest bash`

Then start the MONAI Label server with the model you wish to use. 

You have to set up the Orthanc before: https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/ohifv3#installing-orthanc-dicomweb

For this:

> I am also interested now in testing out the MONAI Model Zoo as well.

You just need to use this configuration to start the MONAI Label server: https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/monaibundle#monai-bundle-application

Also, there is Space in HF available for you to quickly test Orthanc, OHIF and MONAI Label: https://huggingface.co/spaces/MONAI/MONAI-Label-OHIF-Viewer

Hope this helps,
